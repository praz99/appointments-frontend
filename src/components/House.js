import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import '../styles/House.css';

const House = ({ house }) => (
  <div className="house-container" style={{ backgroundImage: 'url(https://www.thecocktaildb.com/images/media/drink/2x8thr1504816928.jpg)' }}>
    <Link to={`/house/${house.id}`} className="house-button">See Details</Link>
  </div>
);

House.propTypes = {
  image: PropTypes.string,
}.isRequired;

export default House;
