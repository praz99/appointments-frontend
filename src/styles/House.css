.house-container {
  height: calc(96.93vh - 105px);
  background: #e5e5f5;
}

.house-image-container {
  height: 85%;

  /* background: rgb(69,64,145);
  background: linear-gradient(90deg, rgba(69,64,145,1) 0%, rgba(229,229,245,1) 5%, rgba(229,229,245,1) 95%, rgba(69,64,145,1) 100%); */
}

.house-image {
  height: 100%;
  object-fit: contain;
}

.house-button-container {
  margin-top: 15px;
}

.house-button {
  background: #e5e5f5;
  border: 1px solid #6f1d1d;
  color: #6f1d1d;
  font-size: 15px;
  margin-top: 10px;
  text-align: center;
  padding: 8px 10px;
  font-weight: 700;
}

.house-button:hover {
  color: #fff;
  background-color: #6f1d1d;
}
