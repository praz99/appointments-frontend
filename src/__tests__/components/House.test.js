import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import House from '../../components/House';

afterEach(cleanup);

const houseNew = {
  id: 1,
  image: 'https://www.thecocktaildb.com/images/media/drink/tqpvqp1472668328.jpg',
};

const houseMount = (
  <MemoryRouter>
    <House house={houseNew} />
  </MemoryRouter>
);

it('renders item component correctly', () => {
  const house = renderer.create(houseMount).toJSON();
  expect(house).toMatchSnapshot();
});

it('renders a link to see house details', () => {
  const { getByTestId } = render(houseMount);
  const link = getByTestId('house-link');
  expect(link.innerHTML).toMatch('See Details');
});
