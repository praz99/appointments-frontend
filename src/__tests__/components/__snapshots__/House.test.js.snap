// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item component correctly 1`] = `
<div
  className="house-container"
>
  <div
    className="house-image-container"
  >
    <img
      className="house-image"
      src="https://www.thecocktaildb.com/images/media/drink/tqpvqp1472668328.jpg"
    />
  </div>
  <div
    className="house-button-container"
  >
    <a
      className="house-button"
      data-testid="house-link"
      href="/houses/1"
      onClick={[Function]}
    >
      See Details
    </a>
  </div>
</div>
`;
